api/src/main/java/net/kyori/adventure/text/ComponentBuilder.java

```diff
@@ -155,6 +156,14 @@
   @Contract("_ -> this")
   @NonNull B mapChildrenDeep(final @NonNull Function<BuildableComponent<?, ?>, ? extends BuildableComponent<?, ?>> function);
 
+  /**
+   * Get an unmodifiable list containing all children currently in this builder.
+   *
+   * @return the list of children
+   * @since 4.6.0
+   */
+  @NonNull List<Component> children();
```

Comment:
Is this method necessary?

Reply:
it's not strictly required -- the collector could just work with the builder implementation -- but it feels weird to not expose functionality the collector needs to the api.

Reply:
I'm not really a fan of exposing getters on builders... but it looks like we do for `content()` as well, so I guess this is okay.